gateleen-routing/src/main/java/org/swisspush/gateleen/routing/RouterConfigurationParser.java

```diff
@@ -0,0 +1,113 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.oauth2.OAuth2FlowType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.swisspush.gateleen.core.util.StringUtils;
+import org.swisspush.gateleen.routing.auth.OAuthConfiguration;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RouterConfigurationParser {
+
+    private static final Logger log = LoggerFactory.getLogger(RouterConfigurationParser.class);
+
+    private static final String REQUEST_HOPS_LIMIT_PROPERTY = "request.hops.limit";
+    private static final String AUTHCONFIGS_PROPERTY = "authConfigs";
+    private static final String FLOWTYPE_PROPERTY = "flowType";
+    private static final String CLIENTID_PROPERTY = "clientId";
+    private static final String CLIENTSECRET_PROPERTY = "clientSecret";
+    private static final String SITE_PROPERTY = "site";
+    private static final String TOKENPATH_PROPERTY = "tokenPath";
+    private static final String AUTHPATH_PROPERTY = "authPath";
+    private static final String SCOPES_PROPERTY = "scopes";
+    private static final String SUPPORTEDGRANTTYPES_PROPERTY = "supportedGrantTypes";
+
+    /**
+     * Parses the provided router configuration resource and returns a {@link RouterConfiguration}.
+     *
+     * @param configurationResourceBuffer the resource to parse
+     * @return a {@link RouterConfiguration}
+     */
+    public static Optional<RouterConfiguration> parse(Buffer configurationResourceBuffer,
+                                                      Map<String, Object> properties) {
+        String replacedConfig;
+        JsonObject config;
+
+        try {
+            replacedConfig = StringUtils.replaceWildcardConfigs(configurationResourceBuffer.toString(UTF_8), properties);
+            config = new JsonObject(Buffer.buffer(replacedConfig));
+        } catch (Exception e) {
+            log.warn("Could not replace wildcards with environment properties for the router configuration " +
+                            "due to following reason: {}", e.getMessage());
+            return Optional.empty();
+        }
+
+        Integer requestHopsLimit = config.getInteger(REQUEST_HOPS_LIMIT_PROPERTY);
+        JsonObject authConfigs = config.getJsonObject(AUTHCONFIGS_PROPERTY);
+        if (authConfigs == null) {
+            return Optional.of(new RouterConfiguration(requestHopsLimit, Collections.emptyMap()));
+        }
+
+        Map<String, OAuthConfiguration> oAuthConfigurationsMap = new HashMap<>();
+
+        for (String authConfigId : authConfigs.fieldNames()) {
+            JsonObject authConfig = authConfigs.getJsonObject(authConfigId);
+
+            String flowTypeStr = authConfig.getString(FLOWTYPE_PROPERTY);
+            OAuth2FlowType flowType = flowTypeFromStr(flowTypeStr);
+
+            if(flowType == null) {
+                log.warn("No valid OAuth2FlowType configured for auth configuration '{}'. " +
```

Comment:
dito

Reply:
The same thing here because only the invalid configuration will not be used and rules using this configuration will also receive failing requests.